torch/_inductor/codegen/cpp.py

```diff
@@ -3032,15 +3124,17 @@ def reduction_acc_type_vec(self, reduction_type, dtype):
             return f"{self._get_mask_type()}"
         return vec_type
 
-    def welford_weight_reciprocal_vec(self, dtype, num_threads=None):
+    def welford_weight_reciprocal_vec(
+        self, dtype, welford_helper_vec_range, welford_helper_val, num_threads=None
+    ):
         vec_num_range_thread = (
-            CeilDiv(self.weight_recp_vec_range, num_threads)
+            CeilDiv(welford_helper_vec_range, num_threads)
             if num_threads
-            else self.weight_recp_vec_range
+            else welford_helper_vec_range
         )
         vec_num_range_thread_expr = cexpr_index(vec_num_range_thread)
         return (
-            f"static WeightRecp<{self._get_vec_type(dtype)}> {self.weight_recps_val}"
+            f"WelfordHelper<{self._get_vec_type(dtype)}> {welford_helper_val}"
```

Comment:
Can we consider using static and thread local for `WelfordHelper` to avoid performance regression?

Reply:
Thanks for your comment! I resized the chunk size to 4096, and used static when the chunk number <=1. I tested Inductor benchmark with this PR, no performance regressions were seen.